File: prop-all/src/test/java/io/pleo/prop/objects/InjectedObjectWithConstructor.java
package io.pleo.prop.objects;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class InjectedObjectWithConstructor {
  private String name;
  private int age;

  @JsonCreator
  public InjectedObjectWithConstructor(@JsonProperty("name") String name, @JsonProperty("age") int age) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getAge() {
    return age;
  }

  public void setAge(int age) {
    this.age = age;
  }
}

File: prop-all/src/test/kotlin/io/pleo/prop/objects/InjectedObjectWithConstructor.kt
package io.pleo.prop.objects

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

@Suppress("unused", "unused_parameter")
class InjectedObjectWithConstructor
    @JsonCreator
    constructor(
        @JsonProperty("name") var name: String,
        @JsonProperty("age") age: Int,
    ) {
        var age = 0
    }
